Lib/multiprocessing/semaphore_tracker.py

```diff
@@ -43,10 +45,16 @@ def ensure_running(self):
         with self._lock:
             if self._pid is not None:
                 # semaphore tracker was launched before, is it still running?
-                pid, status = os.waitpid(self._pid, os.WNOHANG)
-                if not pid:
-                    # => still alive
-                    return
+                try:
+                    pid, _ = os.waitpid(self._pid, os.WNOHANG)
+                except ChildProcessError:
+                    # The process must be death (no process with pid self._pid).
+                    pass
+                else:
+                    if pid is not None:
```

Comment:
I don't think that "pid is not None". I proposed to use "pid is not None" in the test, to test _pid attribute.